dataactbroker/handlers/fileHandler.py

```diff
@@ -479,9 +498,134 @@ def checkD1File(self, submission_id=None):
 
         return JsonResponse.create(StatusCode.OK, response)
 
+    def loadGenerateRequest(self):
+        """ Pull information out of request object and return it
+
+        Returns: tuple of submission ID and file type
+
+        """
+        requestDict = RequestDictionary(self.request)
+        if not (requestDict.exists("submission_id") and requestDict.exists("file_type")):
+            exc = ResponseException("Generate file route requires submission_id and file_type",
+                                    StatusCode.CLIENT_ERROR)
+            return JsonResponse.error(exc, exc.status)
+
+        submission_id = requestDict.getValue("submission_id")
+        file_type = requestDict.getValue("file_type")
+        return submission_id, file_type
+
+    def generateFile(self):
+        """ Start a file generation job for the specified file type """
+        submission_id, file_type = self.loadGenerateRequest()
```

Comment:
Good catch, changed loadGenerateRequest to raise an exception in that case.  Exceptions get translated to the appropriate response object in dataactbroker.routeUtils